### PR TITLE
Fixed #23868 -- Added support for non-unique django-admin options in docs.

### PR DESCRIPTION
**Pull request by timgraham**
Also documented missing short command line options to fix #24134.

https://code.djangoproject.com/ticket/23868
https://code.djangoproject.com/ticket/24134

### CODE REVIEW

File: docs/internals/contributing/writing-documentation.txt [hunk 1]
```diff
@@ -241,7 +241,7 @@ __ http://sphinx-doc.org/markup/
 
         .. django-admin-option:: --traceback
 
-  To link, use ``:djadminopt:`--traceback```.
+  To link, use ``:option:`commandname --traceback```.
```
Comment (charettes) [inline docs/internals/contributing/writing-documentation.txt]: Should we also mention that `commandname` can be omitted if were linking a global option?

File: docs/ref/django-admin.txt [hunk 2]
```diff
@@ -628,33 +640,31 @@ Example usage::
...
 
 .. seealso::
 
     See :ref:`customizing-makemessages` for instructions on how to customize
     the keywords that :djadmin:`makemessages` passes to ``xgettext``.
 
-makemigrations [<app_label>]
-----------------------------
+makemigrations [<app_label> ...]
+--------------------------------
```
Comment (charettes) [inline docs/ref/django-admin.txt]: `loaddata` uses the `[fixture fixture ...]` form.

I personally prefer the `command [<arg> ...]` form better but as long as were consistent I don't mind either.

File: docs/ref/django-admin.txt [hunk 3]
```diff
@@ -244,11 +246,13 @@ example, the default settings don't define :setting:`ROOT_URLCONF`, so
...
-if they have Django's default value. Such settings are prefixed by ``"###"``.
+.. django-admin-option:: --all
 
-dumpdata <app_label app_label app_label.Model ...>
---------------------------------------------------
+Displays all settings, even if they have Django's default value. Such settings
+are prefixed by ``"###"``.
+
+dumpdata <app_label app_label.Model ...>
+----------------------------------------
```
Comment (charettes) [inline docs/ref/django-admin.txt]: Here's another form.

File: docs/ref/django-admin.txt [hunk 4]
```diff
@@ -395,28 +409,30 @@ table's lifecycle, you'll need to change the
...
 it because ``True`` is its default value).
 
-The :djadminopt:`--database` option may be used to specify the
-database to introspect.
+.. django-admin-option:: --database <db_alias>
+
+Specifies the database to introspect. Defaults to ``default``.
 
-loaddata <fixture fixture ...>
+loaddata [fixture fixture ...]
```
Comment (charettes) [inline docs/ref/django-admin.txt]: Are brackets used to denote at least one fixture must be specified?

File: docs/internals/contributing/writing-documentation.txt (hunk 1 above)
Comment (timgraham) [inline docs/internals/contributing/writing-documentation.txt]: Sure

File: docs/ref/django-admin.txt (hunk 4 above)
Comment (timgraham) [inline docs/ref/django-admin.txt]: Brackets are for optional args. I'm roughly copying these from `--help` so this should be `fixture [fixture ...]`. Maybe we should omit the angle brackets as that command does and use things like `--database DATABASE` so there's less translation/thinking involved when writing these up.

**Comment by timgraham** [comment]
Simon, I pushed a second commit (to be squashed) with some updates.

**Comment by charettes** [comment]
I'll have a look in the next 15 min.

**Comment by charettes** [comment]
Except for the plural form inconsistency (unless I'm missing something again) it LGTM.

For example, in the `--locale LOCALE, -l LOCALE` but `--extension EXTENSIONS, -e EXTENSIONS`

**Comment by timgraham** [comment]
This is controlled by the argparse definition, for example. `dest='extensions'` vs. `dest='exclude'`.

**Comment by charettes** [comment]
Alright, good to go then.

**Pull request merged**